### PR TITLE
Update pod charts

### PR DESCRIPTION
**Pull request by rawagner**
JIRA https://issues.redhat.com/browse/CONSOLE-2190

Screenshot from 2020-07-22 10-27-10

Note: the title is aligned to left as can be seen elsewhere, i just didnt update this screenshot
Screenshot from 2020-07-20 15-22-19

If there's just one network we show just typical area chart
Screenshot from 2020-07-20 15-33-25

I've also updated charts in overview pages
Updated tooltip for charts with requested/limit info
Screenshot from 2020-07-22 10-27-54

Updated tooltip for Network Transfer
Screenshot from 2020-07-22 10-28-05

I had to customize the PF ChartLegendTooltip implementation, its mostly a copy paste with a few things changed. There's an open issue for PF to address the customization I had to do https://github.com/patternfly/patternfly-react/issues/4589

@andybraren

### CODE REVIEW

**Comment by andybraren** [comment]
Looks great! And thank you for bringing this better tooltip over to the Node Overview as well. 👍

> changed to stack chart for Network Transfer

The y axis is a little hard to see in the Node Overview screenshot, but since In/Out don't really "add up" to a larger overall number in the same way they do for the separate Pod Details charts, we may not want to stack those measurements in this particular chart. WDYT? Does not stacking like we do today make sense?

**Comment by rawagner** [comment]
> Looks great! And thank you for bringing this better tooltip over to the Node Overview as well. 
> 
> > changed to stack chart for Network Transfer
> 
> The y axis is a little hard to see in the Node Overview screenshot, but since In/Out don't really "add up" to a larger overall number in the same way they do for the separate Pod Details charts, we may not want to stack those measurements in this particular chart. WDYT? Does not stacking like we do today make sense?

Ok, area chart with better tooltip it is. Code & screenshots updated

**Comment by rawagner** [comment]
/assign @spadgett 

PTAL

File: frontend/public/components/pod.tsx [hunk 1]
```diff
@@ -315,6 +316,8 @@ const PodGraphs = requirePrometheus(({ pod }) => (
           byteDataType={ByteDataTypes.BinaryBytes}
           namespace={pod.metadata.namespace}
           query={`sum(container_memory_working_set_bytes{pod='${pod.metadata.name}',namespace='${pod.metadata.namespace}',container='',}) BY (pod, namespace)`}
+          limitQuery={`sum(kube_pod_container_resource_limits_memory_bytes{pod='${pod.metadata.name}',namespace='${pod.metadata.namespace}'}) BY (pod, namespace)`}
+          requestedQuery={`sum(kube_pod_container_resource_requests_memory_bytes{pod='${pod.metadata.name}',namespace='${pod.metadata.namespace}'}) BY (pod, namespace)`}
```
[the remark this answers is not in the document]
Comment (s-urbaniak) [reply frontend/public/components/pod.tsx:320]: lgtm

File: frontend/public/components/graphs/tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,296 @@
+// Mostly just a copy-paste from patternfly.
```
Comment (spadgett) [inline frontend/public/components/graphs/tooltip.tsx:1]: It's too bad you needed to do this.

I realize you need to be able to calculate the total, but are most of the other changes cosmetic? How different does the tooltip look without your changes?

**Comment by spadgett** [comment]
/lgtm cancel
Merge conflict

File: frontend/public/components/graphs/tooltip.tsx (hunk 2 above)
Comment (rawagner) [reply frontend/public/components/graphs/tooltip.tsx:1]: most of the changes are cosmetic. I've created issue at PF https://github.com/patternfly/patternfly-react/issues/4589 where I described all I needed to change

**Comment by spadgett** [comment]
/lgtm cancel

Build error:

```
ERROR in /go/src/github.com/openshift/console/frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/data-consumption-card.tsx
(50,5): Argument of type 'GetRangeStats | ((args: PrometheusResponse) => PrometheusResponse)' is not assignable to parameter of type '(arg: PrometheusResponse) => PrometheusResponse | DataPoint<string | number | Date>[]'.
  Type 'GetRangeStats' is not assignable to type '(arg: PrometheusResponse) => PrometheusResponse | DataPoint<string | number | Date>[]'.
    Type 'DataPoint<Date>[][]' is not assignable to type 'PrometheusResponse | DataPoint<string | number | Date>[]'.
      Type 'DataPoint<Date>[][]' is not assignable to type 'DataPoint<string | number | Date>[]'.
        Type 'DataPoint<Date>[]' has no properties in common with type 'DataPoint<string | number | Date>'.
Child html-webpack-plugin for "index.html":
```

**Comment by rawagner** [comment]
@bipuladh I've tweaked types in `prometheus-hook.ts` PTAL

**Comment by spadgett** [comment]
/lgtm cancel
It looks like there are still build errors.

**Comment by spadgett** [comment]
/lgtm

My fault, stale GitHub status.

**Pull request merged**